openml/utils.py

```diff
@@ -31,6 +33,52 @@
     pass
 
 
+def check_task_existence(task_meta_data: dict) -> Union[int, None]:
+    """Checks if any task with exists on test server that matches the meta data.
+
+    Parameter
+    ---------
+    task_meta_data : dict
+        A dictionary containing meta-information on the task fetched from the test server.
+
+    Return
+    ------
+    int, None
+    """
+    return_val = None
+    try:
+        tasks = openml.tasks.list_tasks(output_format="dataframe")
+        tasks = typing.cast(pd.DataFrame, tasks).loc[
+            tasks["task_type"] == task_meta_data["task_type"]
+        ]
+        if len(tasks) == 0:
+            return None
+        tasks = tasks.loc[tasks.did == task_meta_data["dataset_id"]]
+        if len(tasks) == 0:
+            return None
+        tasks = tasks.loc[tasks.target_feature == task_meta_data["target_name"]]
+        if len(tasks) == 0:
+            return None
+        task_match = []
+        for task_id in tasks.tid.values:
```

Comment:
Please don't use this syntax from pandas, it's deprecated. It should be `tasks['tid'].to_list()`.